charts/argocd-notifications/templates/deployment.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "argocd-notifications.name" . }}-controller
+  labels:
+    {{- include "argocd-notifications.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "argocd-notifications.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "argocd-notifications.selectorLabels" . | nindent 8 }}
+    spec:
```

Comment:
Please add Recreate strategy to make sure two controllers are not sending notifications during upgrade:

```yaml
  strategy:
    type: Recreate
```